### PR TITLE
[11.0][FIX] Make matrix editable depending of the status of the parent

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
It doesn't make sense to always have the matrix in edit mode. Obviously, if you have the matrix in a wizard, it always will be in edit mode. But if you put the matrix in a normal model (not transient), then the expected behavior is that the matrix is editable if you are in edit mode and the matrix is not editable if you are in read mode.

### CODE REVIEW

**Comment by MiquelRForgeFlow** [comment]
@grindtildeath @simahawk @lreficent @jbeficent

**Comment by JordiBForgeFlow** [comment]
cc @hbrunn

File: web_widget_x2many_2d_matrix/static/src/js/2d_matrix_renderer.js [hunk 1]
```diff
@@ -238,7 +238,7 @@ odoo.define('web_widget_x2many_2d_matrix.X2Many2dMatrixRenderer', function (requ
       if (modifiers.invisible && !(options && options.renderInvisible)) {
           return $td;
       }
-      options.mode = 'edit';  // enforce edit mode
+      options.mode = this.__parentedParent.mode;  // enforce mode of the parent
```
Comment (simahawk) [inline web_widget_x2many_2d_matrix/static/src/js/2d_matrix_renderer.js]: wouldn't be better to use `this.getParent().mode`?

**Review by simahawk — CHANGES_REQUESTED** [review]
also, version update missing

**Review by simahawk — APPROVED** [review]

**Review by hbrunn — APPROVED** [review]
travis failure is unrelated

**Comment by JordiBForgeFlow** [comment]
Thanks! We have a nice surprise coming from @mreficent shortly..

**Pull request merged**